### PR TITLE
Treat backend warnings as errors

### PR DESCRIPTION
**Pull request by johnthagen**
- Treat all warnings as errors so that warnings cannot slip into `master`
- Suppress two `async` related warnings locally

This change is [Reviewable](https://reviewable.io/reviews/sillsdev/thecombine/342)

### CODE REVIEW

File: Backend/Controllers/AudioController.cs [hunk 1]
```diff
@@ -34,7 +34,9 @@ public AudioController(IWordRepository repo, IWordService wordService, IPermissi
         /// <returns> Audio file stream </returns>
         [AllowAnonymous]
         [HttpGet("{wordId}/download/audio/{fileName}")]
+#pragma warning disable 1998
```
Comment (jasonleenaylor) [inline Backend/Controllers/AudioController.cs:37]: When we do this I prefer to also have a comment that explains why the compiler was wrong

Comment (johnthagen) [reply Backend/Controllers/AudioController.cs:37]: @jasonleenaylor Part of what I wanted to discuss is if there was a way to `await` some of the operations in these functions. I'll do some digging today to see if either of these functions can be made properly `async`.

Comment (johnthagen) [reply Backend/Controllers/AudioController.cs:39]: @jasonleenaylor Comments have been added.

**Comment by imnasnainaec** [comment]
This resolves #299

**Pull request merged**